storage-api/src/java/org/apache/hadoop/hive/ql/io/sarg/SearchArgumentImpl.java

```diff
@@ -132,6 +130,14 @@ public Object getLiteral() {
       return literalList;
     }
 
+    public int getId() {
```

Comment:
I believe **Id** setter and getter could be moved to the **PredicateLeaf** interface.
It would be cleaner and as a result **addLeaf** would need less modifications.

Reply:
I made getId more visible, but left setId as private. setId only makes sense in the context of the builder.